pkg/daemon/daemon.go

```diff
@@ -2889,3 +2890,43 @@ func (dn *Daemon) getUnsupportedPackages() {
 	}
 	unsupportedPackages.WithLabelValues(dn.name).Set(float64(unsupportedPackageCount))
 }
+
+func (dn *Daemon) preserveDaemonLogs() {
+	// Find current daemon log location
+	var currentLogLocation, logDir string
+	if walkErr := filepath.WalkDir("/var/log/pods/", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if strings.Contains(d.Name(), "openshift-machine-config-operator_machine-config-daemon-") {
+			currentLogLocation = path
+			logDir = d.Name()
+		}
+		return nil
+	}); walkErr != nil {
+		klog.Errorf("Daemon logs could could not be found due to error: %v", walkErr)
+		return
+	}
+
+	baseDir := "/etc/machine-config-daemon/previous-logs/"
+	// Remove the old logs if it exists
+	if err := os.RemoveAll(baseDir); err != nil {
+		klog.Errorf("Failed to clear out old daemon logs: %v", err)
+		return
+	}
+
+	// Ensure the base directory exists
+	if err := os.MkdirAll(baseDir, 0755); err != nil {
+		klog.Errorf("Failed to create logs directory: %v", err)
+		return
+	}
+
+	// Create preservation location; this will change based on the daemon name/uid
+	preservationLocation := baseDir + logDir
+	cmd := exec.Command("cp", "--recursive", currentLogLocation, preservationLocation)
```

Comment:
Easier to do a recursive file copy via shell command instead of walking the directory, so I went this route.